integration/test_fulltext.py

```diff
@@ -1402,6 +1407,84 @@ def test_proximity_inorder_compat(self):
             result = client.execute_command("FT.SEARCH", "idx", 'apple (yellow green orange one | yellow green orange one) purple (ten | ten) one', "INORDER")
             assert result[0] == 0
 
+    def test_fuzzy_search(self):
+        client: Valkey = self.server.get_new_client()
+        # Create index with text fields
+        client.execute_command("FT.CREATE", "idx1", "ON", "HASH", "SCHEMA",
+                            "content", "TEXT", "NOSTEM")
+        client.execute_command("FT.CREATE", "idx2", "ON", "HASH", "SCHEMA",
+                    "content", "TEXT")
+        client.execute_command("FT.CREATE", "idx3", "ON", "HASH", "SCHEMA",
+                    "content", "TEXT", "NOSTEM", "content2", "TEXT", "NOSTEM")
+        client.execute_command("HSET", "doc:1", "content", "I am going to a race")
+        client.execute_command("HSET", "doc:2", "content", "Carrie needs to take care")
+        client.execute_command("HSET", "doc:3", "content", "who is driving?")
+        client.execute_command("HSET", "doc:4", "content", "Driver drove the car!")
+        # Wait for index backfill to complete
+        IndexingTestHelper.wait_for_backfill_complete_on_node(client, "idx1")
+        # Simple Edit distance (ED) = 1
+        result = client.execute_command("FT.SEARCH", "idx1", '%car%')
+        assert (result[0], set(result[1::2])) == (2, {b"doc:2", b"doc:4"})
+        # Should be case insensitive
+        result = client.execute_command("FT.SEARCH", "idx1", '%CAR%')
+        assert (result[0], set(result[1::2])) == (2, {b"doc:2", b"doc:4"})
+        # Transposition (Damerau-Levenshtein) ED = 1
+        result = client.execute_command("FT.SEARCH", "idx1", '%crA%')
+        assert (result[0], set(result[1::2])) == (1, {b"doc:4"})
+        client.execute_command("HSET", "doc:5", "content", "abdc")
+        result = client.execute_command("FT.SEARCH", "idx1", '%%bacd%%')
+        # race from doc:1 (ED = 2) and abcd from doc:4 (ED = 2, transposition)
+        assert (result[0], set(result[1::2])) == (2, {b'doc:1', b'doc:5'})
+        # In Composed AND
+        client.execute_command("HSET", "doc:11", "content", "Carl Weathers drove the big tramcar")
+        IndexingTestHelper.wait_for_backfill_complete_on_node(client, "idx1")
+        result = client.execute_command("FT.SEARCH", "idx1", 'Driver drove the %Kar%')
+        assert (result[0], set(result[1::2])) == (1, {b"doc:4"})
+        result = client.execute_command("FT.SEARCH", "idx1", 'drove the %car%')
+        assert (result[0], set(result[1::2])) == (2, {b"doc:4", b"doc:11"})
+        # Test with slop
+        result = client.execute_command("FT.SEARCH", "idx1", 'drove the %car%', "SLOP", "0")
+        assert (result[0], set(result[1::2])) == (1, {b"doc:4"})
+        # Test with Inorder
+        result = client.execute_command("FT.SEARCH", "idx1", 'drove the %car%', "INORDER")
```

Comment:
nit: it might be more interesting if the tests stress the SLOP and INORDER features a bit more. You can start with a composed AND that only returns a result without them. Then test that the search fails with SLOP or INORDER and add a new search that does return a result.